test/test_server.py

```diff
@@ -316,6 +317,14 @@ class TestHTTPS(tservers.HTTPProxTest, CommonMixin, TcpMixin):
     clientcerts = True
 
     def test_clientcert(self):
+        self.config.clientcerts = os.path.join(
+            tutils.test_data.path("data/clientcert"), "client.pem")
+        f = self.pathod("304")
+        assert f.status_code == 304
+        assert self.server.last_log()["request"]["clientcert"]["keyinfo"]
+
+    def test_clientcerts(self):
+        self.config.clientcerts = tutils.test_data.path("data/clientcert")
```

Comment:
> But, this also meant a re-running of all of the tests that belong to the classes that TestHTTPS inherits from (tservers.HTTPProxtTest, CommonMixin, TcpMixin).

FWIW, the tests are only in CommonMixin and TcpMixin, so just inherting from tservers.HTTPProxtTest would have worked. However, I like your current approach better, so that doesn't matter in the end. :relaxed: 

> I've checked to make sure this logic is not used elsewhere. So, if you agree, I'll also remove this cruft from the ProxTestBase.

Yes, please! In fact, my previous comment linked to the place in ProxTestBase with the intent for it to be removed as well. Sorry if that was not clear. :smiley: